esp8266-fastled-webserver/include/configs/product/product_template.h

```diff
@@ -0,0 +1,174 @@
+/*
```

Comment:
SIMPLE FILE REVIEW

Each file here defines the options that can vary between the different
products.    The following is the list of symbols currently defining a
given product's configuration (most are self-explanatory):

* `LED_TYPE` -- FastLED `CPixelLEDController` type, such as `WS2811`
* `COLOR_ORDER` -- FastLED `RGB_ORDER` type
* `NUM_PIXELS` -- integer
* `AVAILABLE_MILLI_AMPS` -- integer
* `MAX_MILLI_AMPS_PER_PIXEL` -- integer
* `FRAMES_PER_SECOND` -- integer
* `DEFAULT_PATTERN_INDEX` -- integer
* `DEFAULT_BRIGHTNESS_INDEX` -- integer
* `DEFAULT_COLOR_CORRECTION` -- FastLED `LEDColorCorrection` type
* `NAME_PREFIX` -- Used for auto-generated hostname prefix
* `PRODUCT_FRIENDLY_NAME` -- Used for HTML page title, navigation bar (changed via app.js using json data from device)
* `IS_FIBONACCI` -- 0/1
* `HAS_COORDINATE_MAP` -- 0/1 -- only currently false for default and ESPTHING
* `HAS_POLAR_COORDINATES` -- 0/1 -- only currently defined for 1628Rings product.  I intend to rename this, as its name is misleading.
* `DATA_PIN` -- optional override of device pin -- uses board default if not defined
* `PARALLEL_OUTPUT_CHANNELS` -- optional integer (default: 1) -- Currently only Fib512 (4) and ESPTHING (6) are not set to 1.
  * `PIXELS_ON_DATA_PIN_n` -- only when `PARALLEL_OUTPUT_CHANNELS` is defined, this must define the number of pixels for each channel
  * `DATA_PIN_n` -- optional override of device pin for each pins  `n = [2 .. (PARALLEL_OUTPUT_CHANNELS-1)]`, only when `PARALLEL_OUTPUT_CHANNELS` is defined
* `ENABLE_IR` -- definition of this symbol (even if defined as `0`) enables IR support.  NOTE: No product currently enables this by default.
  * `IR_RECV_PIN` -- optional overrride of device pin for IR receiver -- uses board default if not defined